src/mgmt/cli_mgmt.c

```diff
@@ -380,22 +380,29 @@ int dc_mgmt_net_cfg(const char *name)
 	}
 
 	ofi_interface = getenv("OFI_INTERFACE");
+	ofi_domain = getenv("OFI_DOMAIN");
 	if (!ofi_interface) {
 		rc = setenv("OFI_INTERFACE", info.interface, 1);
 		if (rc != 0)
 			D_GOTO(cleanup, rc = d_errno2der(errno));
-	} else {
-		D_INFO("Using client provided OFI_INTERFACE: %s\n",
-			ofi_interface);
-	}
 
-	ofi_domain = getenv("OFI_DOMAIN");
-	if (!ofi_domain) {
+		/*
+		 * If we use the agent as the source, client env shouldn't be allowed to override
+		 * the domain. Otherwise we could get a mismatch between interface and domain.
+		 */
+		if (ofi_domain)
+			D_WARN("Using agent for network config. Ignoring client provided "
+			       "OFI_DOMAIN: %s\n", ofi_domain);
+
 		rc = setenv("OFI_DOMAIN", info.domain, 1);
 		if (rc != 0)
 			D_GOTO(cleanup, rc = d_errno2der(errno));
 	} else {
-		D_INFO("Using client provided OFI_DOMAIN: %s\n", ofi_domain);
+		D_INFO("Using client provided OFI_INTERFACE: %s\n", ofi_interface);
```

Comment:
While we're at it, I think we should either make these debugs or get rid of them. I would be fine with a single INFO at the end to show how the client will be configured without exposing the implementation details, e.g. `D_INFO("Selected interface: %s; domain: %s\n", ...);`

Reply:
I've been finding them pretty useful for debug (for this issue, for instance). On those grounds I think it's worthwhile to leave them in at the info level. They're not especially noisy, either-- they only happen at the beginning of the client log.